core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/DeleteVolumesActions.java

```diff
@@ -0,0 +1,91 @@
+import static com.sequenceiq.cloudbreak.core.flow2.cluster.deletevolumes.DeleteVolumesEvent.DELETE_VOLUMES_HANDLER_EVENT;
+import static com.sequenceiq.cloudbreak.core.flow2.cluster.deletevolumes.DeleteVolumesEvent.DELETE_VOLUMES_VALIDATION_HANDLER_EVENT;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_DELETEVOLUMES_FAILED;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_DELETINGVOLUMES;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.StackDeleteVolumesRequest;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.AbstractClusterAction;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.ClusterViewContext;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.verticalscale.CoreVerticalScaleEvent;
+import com.sequenceiq.cloudbreak.core.flow2.event.DeleteVolumesTriggerEvent;
+import com.sequenceiq.cloudbreak.core.flow2.stack.AbstractStackFailureAction;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.core.flow2.stack.StackFailureContext;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackFailureEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.DeleteVolumesHandlerRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.DeleteVolumesRequest;
+
+@Configuration
+public class DeleteVolumesActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteVolumesActions.class);
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Bean(name = "DELETE_VOLUMES_VALIDATION_STATE")
+    public Action<?, ?> deleteVolumesValidationAction() {
+        return new AbstractClusterAction<>(DeleteVolumesTriggerEvent.class) {
+
+            @Override
+            protected void doExecute(ClusterViewContext context, DeleteVolumesTriggerEvent payload, Map<Object, Object> variables) {
+                StackDeleteVolumesRequest stackDeleteVolumesRequest = payload.getStackDeleteVolumesRequest();
+                flowMessageService.fireEventAndLog(stackDeleteVolumesRequest.getStackId(), Status.UPDATE_IN_PROGRESS.name(), CLUSTER_DELETINGVOLUMES,
+                        stackDeleteVolumesRequest.getGroup(), stackDeleteVolumesRequest.getStackId().toString());
+                String selector = DELETE_VOLUMES_VALIDATION_HANDLER_EVENT.event();
+                DeleteVolumesTriggerEvent validationHandlerEvent = new DeleteVolumesTriggerEvent(selector, payload.getResourceId(), stackDeleteVolumesRequest);
```

Comment:
It is suspicious that the accepted and the sent event of the action is the same `DeleteVolumesTriggerEvent`. Can we have a separate one which will processed by the handler?